guacamole/src/main/webapp/app/client/controllers/clientController.js

```diff
@@ -219,6 +220,12 @@ angular.module('client').controller('clientController', ['$scope', '$routeParams
         remaining: 15
     };
 
+    // Catch window or tab closing (ctrl-w) and prompt the user
+    var ctrlwlistener = function onBeforeUnload(e) {
+            e.preventDefault();
+            e.returnValue = '';
+     };
```

Comment:
1. Please correct the indentation here. The body of the function appears to be indented far further than elsewhere (we use 4-space indents for each level), and the closing brace of the function definition is misaligned.
2. Please use JSDoc to document functions, their return values, and their parameters.
3. I think calling this function `ctrlwlistener()` is a bit misleading. Not all browsers will use Ctrl+W for closing a tab or window, nor is this function directly related to that shortcut.

Reply:
Thanks, will look at it.